plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -122,6 +125,34 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
         return new DynamicTable(pinotTableName, suffix, selectionColumns, groupByColumns, filter, aggregationExpressionBuilder.build(), orderBy, getTopNOrLimit(request), getOffset(request), query);
     }
 
+    public static Optional<String> getAggregationFunctionName(AggregateFunction aggregateFunction)
+    {
+        requireNonNull(aggregateFunction, "aggregateFunction is null");
+        if (aggregateFunction.isDistinct()) {
+            if (aggregateFunction.getFunctionName().equalsIgnoreCase("count")) {
+                // When aggregation expressions push down is implemented,
+                // count(distinct <expression>) will be mapped to distinctcount(expression)
+                // See https://github.com/trinodb/trino/issues/4171
+                // Until then this branch will never be executed
+                return Optional.of("distinctcount");
+            }
+            return Optional.empty();
+        }
+        return Optional.of(aggregateFunction.getFunctionName());
+    }
+
+    public static String getInputExpresion(AggregateFunction aggregateFunction)
+    {
+        requireNonNull(aggregateFunction, "aggregateFunction is null");
+        Optional<String> aggregationFunctionName = getAggregationFunctionName(aggregateFunction);
+        boolean isCountFunction = aggregationFunctionName.map(name -> name.equalsIgnoreCase("count")).orElse(false);
+        // Pinot converts count(column_name) to count(*)
```

Comment:
The two are not equivalent. in SQL, `count(column_name)` counts how many non-null values there are

Reply:
Thanks! I will add the is not null filter for that case.

Reply:
Changed to preserve the column name. Null value support will then be handled by pinot (still in progress): https://docs.pinot.apache.org/developers/advanced/null-value-support.

Reply:
Update: pinot converts count(<column name>) to count(*). Changed behavior to enforce that only count(*) is used to avoid result set column mapping mismatches. When count(<column name>) is supported in pinot I will update the code.

Reply:
> `Update: pinot converts count(<column name>) to count(*).`

We should have a correctness-ensuring, reusable connector test for `count(a)` over nullable data.
Can you please ensure we have one, or add?

Reply:
I added tests to ensure count(*) is used, and enabled the "null handling" pinot config on the test table. There is no null value in pinot, i.e. string null is 'null', long null becomes Long.MIN_VALUE, int null -> Integer.MIN_VALUE.

Reply:
Added thorough null tests and other behavior. Also extracted the first 2 commits to #7627